src/rest/json.cpp

```diff
@@ -0,0 +1,559 @@
+
+    return json;
+}
+
+cJSON *Route2Json(const otNetworkDiagRoute &aRoute)
+{
+    cJSON *json = cJSON_CreateObject();
+
+    cJSON_AddItemToObject(json, "IdSequence", cJSON_CreateNumber(aRoute.mIdSequence));
+
+    cJSON *RouteData = cJSON_CreateArray();
+    for (uint16_t i = 0; i < aRoute.mRouteCount; ++i)
+    {
+        cJSON *RouteDatavalue = RouteData2Json(aRoute.mRouteData[i]);
+
+        cJSON_AddItemToArray(RouteData, RouteDatavalue);
+    }
+
+    cJSON_AddItemToObject(json, "RouteData", RouteData);
+
+    return json;
+}
+
+cJSON *LeaderData2Json(const otLeaderData &aLeaderData)
+{
+    cJSON *json = cJSON_CreateObject();
+
+    cJSON_AddItemToObject(json, "PartitionId", cJSON_CreateNumber(aLeaderData.mPartitionId));
+
+    cJSON_AddItemToObject(json, "Weighting", cJSON_CreateNumber(aLeaderData.mWeighting));
+
+    cJSON_AddItemToObject(json, "DataVersion", cJSON_CreateNumber(aLeaderData.mDataVersion));
+
+    cJSON_AddItemToObject(json, "StableDataVersion", cJSON_CreateNumber(aLeaderData.mStableDataVersion));
+
+    cJSON_AddItemToObject(json, "LeaderRouterId", cJSON_CreateNumber(aLeaderData.mLeaderRouterId));
+
+    return json;
+}
+
+std::string IpAddr2JsonString(const otIp6Address &aAddress)
+{
+    std::string str;
+    uint16_t    value;
+
+    char p[5];
+
+    for (int i = 0; i < 16; ++i)
+    {
+        value = aAddress.mFields.m8[i];
+        if (i % 2 == 1)
+        {
+            value = (aAddress.mFields.m8[i - 1] << 8) + aAddress.mFields.m8[i];
+            sprintf(p, "%x", value);
+            if (i > 1)
+            {
+                str += ":";
+            }
+            str += std::string(p);
+        }
+    }
+    return String2JsonString(str);
+}
+
+cJSON *Bytes2HexJson(const uint8_t *aBytes, uint8_t aLength)
+{
+    char hex[2 * aLength + 1];
+
+    cJSON *json;
+
+    otbr::Utils::Bytes2Hex(aBytes, aLength, hex);
+
+    hex[2 * aLength] = '\0';
+
+    json = cJSON_CreateString(hex);
+
+    return json;
+}
+
+std::string Node2JsonString(Node node)
+{
+    cJSON *json = cJSON_CreateObject();
+    cJSON *value;
+
+    value = cJSON_CreateNumber(node.role);
+    cJSON_AddItemToObject(json, "State", value);
+
+    value = cJSON_CreateNumber(node.numOfRouter);
+    cJSON_AddItemToObject(json, "NumOfRouter", value);
+
+    value = IpAddr2Json(node.rlocAddress);
+    cJSON_AddItemToObject(json, "RlocAddress", value);
+
+    value = Bytes2HexJson(node.extAddress, OT_EXT_ADDRESS_SIZE);
+    cJSON_AddItemToObject(json, "ExtAddress", value);
+
+    value = CString2Json(node.networkName);
+    cJSON_AddItemToObject(json, "NetworkName", value);
+
+    value = cJSON_CreateNumber(node.rloc16);
+    cJSON_AddItemToObject(json, "Rloc16", value);
+
+    value = LeaderData2Json(node.leaderData);
+    cJSON_AddItemToObject(json, "LeaderData", value);
+
+    value = Bytes2HexJson(node.extPanId, OT_EXT_PAN_ID_SIZE);
+    cJSON_AddItemToObject(json, "ExtPanId", value);
+
+    std::string ret = Json2String(json);
```

Comment:
Do we put all variable declarations together ? @wgtdkp  @tttttangTH

Reply:
Yes, it should be, I forgot it here.